ctlearn/image_mapping.py

```diff
@@ -648,67 +713,76 @@ def generate_table_HESS(self, tel_type):
+        pos_int = pos / pixel_length * 2
+        pos_int[0, :] = pos_int[0, :] / np.sqrt(3) * 2
+        # below put the image in the corner
+        pos_int[0, :] -= np.min(pos_int[0, :])
+        pos_int[1, :] -= np.min(pos_int[1, :])
+        p0_lim = np.max(pos_int[0, :]) - np.min(pos_int[0, :])
+        p1_lim = np.max(pos_int[1, :]) - np.min(pos_int[1, :])
+        if output_dim < p0_lim or output_dim < p1_lim:
+            print("Danger! output image shape too small, will be cropped!")
+        # below put the image in the center
+        if tel_type in ["FACT"]:
+            pos_int[0, :] += (output_dim - p0_lim) / 2. - 1.0
+            pos_int[1, :] += (output_dim - p1_lim - 0.8) / 2. - 1.0
+        else:
+            pos_int[0, :] += (output_dim - p0_lim) / 2.
+            pos_int[1, :] += (output_dim - p1_lim - 0.8) / 2.
+
+
+        mapping_matrix = np.zeros((num_pixels + 1, output_dim, output_dim), dtype=float)
             
-            for i in range(num_pixels):
-                x, y = pos_int[:, i]
-                x_S = int(round(x))
-                x_L = x_S + 1
-                y_S = int(round(y))
-                y_L = y_S + 1
-                # leave 0 for padding, mapping matrix from 1 to 499
-                #mapping_matrix[i + 1, x_S:x_L + 1, y_S:y_L + 1] = pixel_weight
-                mapping_matrix[i + 1, y_S:y_L + 1, x_S:x_L + 1] = pixel_weight
-
-            # make sparse matrix of shape (num_pixels + 1, output_dim * output_dim)
-            mapping_matrix = csr_matrix(mapping_matrix.reshape(num_pixels + 1, output_dim * output_dim))
+        for i in range(num_pixels):
+            x, y = pos_int[:, i]
+            x_S = int(round(x))
+            x_L = x_S + 1
+            y_S = int(round(y))
+            y_L = y_S + 1
+            # leave 0 for padding, mapping matrix from 1 to 499
+            #mapping_matrix[i + 1, x_S:x_L + 1, y_S:y_L + 1] = pixel_weight
+            mapping_matrix[i + 1, y_S:y_L + 1, x_S:x_L + 1] = pixel_weight
 
-        else:
-            raise NotImplementedError("Cannot convert hexagonal camera image without valid conversion algorithm.")
+        # make sparse matrix of shape (num_pixels + 1, output_dim * output_dim)
+        mapping_matrix = csr_matrix(mapping_matrix.reshape(num_pixels + 1, output_dim * output_dim))
 
         return mapping_matrix
 
+    def slice_pixelPos(self, pos, num_pixels):
+        slice_pos = []
+        slice_pos.append(pos[0][0:num_pixels])
+        slice_pos.append(pos[1][0:num_pixels])
+        slice_pos=np.array(slice_pos)
+        return slice_pos
+        
+    '''
+    The pixel positions from files are already rotated!
     def rotate_cam(self, pos):
         rotation_matrix = np.matrix([[0.98198181, 0.18897548],
                              [-0.18897548, 0.98198181]], dtype=float)
         pos_rotated = np.squeeze(np.asarray(np.dot(rotation_matrix, pos)))
 
         return pos_rotated
-
+    '''
+    
     def rebinning(self):
```

Comment:
This function isn't used at all, not sure where it came from. Unless @qi-feng @bryankim96 have some objection, let's just delete it.

Reply:
Yes, we can delete it.

Reply:
We can discuss it in the next call. I modified the pixel position files in `ctlearn/ctlearn/pixel_pos_files/` so that we don't need this function anymore.